docs/decisions/0002-auto-enrollment-post-request-approval.rst

```diff
@@ -0,0 +1,158 @@
+After a request is approved, the learner has to redeem the subsidy and enroll in a course manually.
+Since the goal of a learner is to access a course, we want to automatically enroll them after a request approval.
+
+For coupon code requests, requiring manual redemption is not ideal because we cannot enforce that a code is only used
+for the course it was requested for.
+Admins have some visibility into the usage of a coupon code via the admin portal. It shows only if a coupon code has been redeemed,
+not the course it's was redeemed for. We have to query ecommerce data to check the order created using the coupon code
+for that information. Automatically enrolling a learner after a request approval would guarantee that a code is used to enroll in
+the intended course.
+
+For license requests, it's less of a problem because a license can be used for all courses in an enterprise catalog and
+auto-enrollment would just save learners clicks for this case.
+
+Questions to ponder (this is a draft so I'm putting it here for visibility for now):
+* Does enrolling a learner after a request approval provide a good/better user experience?
+  The current flow for request approval gives learners the flexibility to choose when to enroll in a course.
+  A user might request access to multiple courses with the intention of taking them one at a time or sometime later.
+  Automatically enrolling them in multiple courses might force learners to bookmark courses or even unenroll.
+* How valuable is the guarantee that a code is used to enroll a learner in the intended course?
+  A learner can use a code intended for one course to enroll in another, but it's an edge case since the email we send
+  after a request approval links them to the request course.
+* Which course run do we enroll users in if there are multiple? Do we allow learners to choose or do we enroll them in the first available course run?
+* How does this impact analytics/downstream reporting? We need to provide a way for analytics to separate auto enrollments from regular enrollments.
+
+Decision
+============
+
+We're attempting to adopt event-based communication between the enterprise-access service, the LMS (possibly), and the services
+responsible for granting/redeeming a subsidy.
+
+The following events will be sent to an event bus when the state of a subsidy request is approved:
+* `org.openedx.enterprise.access.coupon-code-request.approved.v1`
+* `org.openedx.enterprise.access.license-request.approved.v1``
+These events will be produced via Django signals that live in the `openedx-events <https://github.com/openedx/openedx-events>`__ repo.
+
+Auto-enrollment will be a feature that could be toggled off as it does not always make sense to auto-enroll a learner,
+e.g. when a coupon code offers 50% discount, a license request without course id, etc.. We will add a field `attempt_auto_enrollment`
```

Comment:
Question/thoughts for the team:

Case: customer has auto-enroll on. customer has 1 coupon. is 50% off. user requests course in coupon catalog... can admin even approve the request? also, does admin control toggle for whether or not they're on auto approval?

Case: customer has auto-enroll on. customer has 3 coupons. 2 coupons are 100%, one is 50%. a course exists in catalogs for 50% and 100% coupons. what's the UX on approval for the admin?